services/gateway/shared/ready.js

```diff
@@ -16,56 +16,52 @@
 const BluebirdPromise = require('bluebird');
 const { MoleculerError } = require('moleculer').Errors;
 
+const logger = require('lisk-service-framework').Logger();
+const { getAppContext } = require('./appContext');
 const config = require('../config');
 
-const currentStatus = {
-	indexReadyStatus: false,
-	transactionStatsStatus: false,
-	feesStatus: false,
-	delegatesStatus: false,
+const currentSvcStatus = {
+	indexer: false,
+	connector: false,
+	fees: false,
+	market: false,
+	newsfeed: false,
+	'app-registry': false,
+	statistics: false,
 };
 
-const getReady = async broker => {
-	const coreMethods = {
-		lisk_accounts: 'core.accounts',
-		lisk_blocks: 'core.blocks',
-		lisk_transactions: 'core.transactions',
-		// lisk_peers: 'core.peers',
-	};
-	try {
-		const services = await BluebirdPromise.map(
-			Object.getOwnPropertyNames(coreMethods),
-			async key => {
-				const service = {};
-				const response = await broker.call(coreMethods[key], { limit: 10 });
-				service[key] = !!response.data.length;
-				return service;
-			},
-		);
-		let allServices;
-		if (config.includeCoreReadiness) {
-			allServices = Object.assign(...services, currentStatus);
-		} else {
-			allServices = Object.assign(...services);
-		}
+const updateSvcStatus = async () => {
+	await BluebirdPromise.map(
+		Object.keys(currentSvcStatus),
+		async microservice => {
+			const broker = (await getAppContext()).getBroker();
+			currentSvcStatus[microservice] = await broker.call(`${microservice}.status`)
+				.then((res) => res.isReady)
+				.catch((err) => {
+					logger.error(err);
+					return false;
+				});
+		},
+	);
+};
 
-		const servicesStatus = !Object.keys(allServices).some(value => !allServices[value]);
-		if (servicesStatus) return Promise.resolve({ services: allServices });
-		return Promise.reject(new MoleculerError('Core Service Unavailable', 503, 'CORE_SERVICES_NOT_READY', currentStatus));
+const getReady = async () => {
+	const includeSvcForReadiness = Object.entries(currentSvcStatus)
+		.filter(([key]) => config.brokerDependencies.includes(key));
+	const includeSvcForReadinessObj = Object.fromEntries(includeSvcForReadiness);
+	try {
+		const servicesStatus = !Object.keys(includeSvcForReadinessObj)
+			.some(value => !includeSvcForReadinessObj[value]);
```

Comment:
What are we trying to achieve here?
```suggestion
		const servicesStatus = !Object.keys(includeSvcForReadinessObj)
			.some(key => !includeSvcForReadinessObj[key]);
```